src/hooks/core/chat/useSystemMessage.ts

```diff
@@ -111,21 +111,21 @@ const useSystemMessage = () => {
 
   function getFirstMessage(messages: GetstreamMessage[]): GetstreamMessage | null | undefined {
     if (!messages) return null;
-    if (messages.length === 0) return null;
+    if (messages?.length === 0) return null;
     const sortedMessages = [...messages]?.sort((a, b) =>
-      moment(b?.updated_at).diff(moment(a?.updated_at))
+      moment(b?.updated_at)?.diff(moment(a?.updated_at))
     );
 
     let firstMessage: GetstreamMessage | null | undefined = null;
     while (sortedMessages?.length > 0) {
-      const checkFirstMessage = sortedMessages?.at(0);
+      const checkFirstMessage = sortedMessages[0];
       if (!__isSystemMessage(checkFirstMessage)) {
         firstMessage = checkFirstMessage;
         break;
       }
 
       if (checkFirstMessage?.text?.toLocaleLowerCase()?.includes('this topic has new')) {
-        sortedMessages.shift();
+        sortedMessages?.shift();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of optional chaining (`?.`) is unnecessary in some places. For instance, on line 115, you're already checking if `messages` is not null or undefined before this point, so there's no need to use optional chaining again. Similarly, on line 121, you don't need to use optional chaining with `sortedMessages` because it will always be an array (possibly empty) due to the sorting operation on line 116. The same applies to line 128.

```diff
- const sortedMessages = [...messages]?.sort((a, b) =>
+ const sortedMessages = [...messages].sort((a, b) =>
- while (sortedMessages?.length > 0) {
+ while (sortedMessages.length > 0) {
- sortedMessages?.shift();
+ sortedMessages.shift();
```

154:
The use of optional chaining (`?.`) is unnecessary here. As mentioned above, `sortedMessages` will always be an array (possibly empty), so there's no need to use optional chaining.

```diff
- sortedMessages?.shift();
+ sortedMessages.shift();
```

173:
Again, the use of optional chaining (`?.`) is unnecessary here. You can safely call `shift()` directly on `sortedMessages`.

```diff
- sortedMessages?.shift();
+ sortedMessages.shift();
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->